frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,625 @@
+                name: "Create a new dashboard",
+              }),
+            ).toBeInTheDocument();
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: PERSONAL_SUBCOLLECTION.name,
+              }),
+            );
+
+            expect(
+              screen.getByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).toBeInTheDocument();
+          });
+        });
+
+        describe("whether we should render dashboards in a collection", () => {
+          it("should render dashboards when opening the root collection (public collection)", async () => {
+            // no `collection` and `collection_id` means it's in the root collection
+            const dashboardInRootCollection = createMockDashboard({
+              id: 5,
+              name: "Dashboard in root collection",
+              model: "dashboard",
+            });
+
+            await setup({
+              dashboard: dashboardInRootCollection,
+            });
+
+            expect(
+              await screen.findByRole("heading", {
+                name: dashboardInRootCollection.name,
+              }),
+            ).toBeInTheDocument();
+          });
+
+          it("should render dashboards when opening public subcollections", async () => {
+            const dashboardInPublicSubcollection = createMockDashboard({
+              id: 5,
+              name: "Dashboard in public subcollection",
+              collection_id: COLLECTION.id as number,
+              model: "dashboard",
+            });
+
+            await setup({
+              dashboard: dashboardInPublicSubcollection,
+            });
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: COLLECTION.name,
+              }),
+            );
+
+            expect(
+              await screen.findByRole("heading", {
+                name: dashboardInPublicSubcollection.name,
+              }),
+            ).toBeInTheDocument();
+          });
+
+          it("should render dashboards when opening personal collections", async () => {
+            const dashboardInPersonalCollection = createMockDashboard({
+              id: 5,
+              name: "Dashboard in personal collection",
+              collection_id: PERSONAL_COLLECTION.id as number,
+              model: "dashboard",
+            });
+
+            await setup({
+              dashboard: dashboardInPersonalCollection,
+            });
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: PERSONAL_COLLECTION.name,
+              }),
+            );
+
+            expect(
+              await screen.findByRole("heading", {
+                name: dashboardInPersonalCollection.name,
+              }),
+            ).toBeInTheDocument();
+          });
+
+          it("should render dashboards when opening personal subcollections", async () => {
```

Comment:
test Query builder > Newly created question > #4 from [the epic](https://github.com/metabase/metabase/issues/34733)